assembly/compliance-tool/src/main/java/org/eclipse/sw360/antenna/frontend/compliancetool/sw360/exporter/SW360Exporter.java

```diff
@@ -49,7 +53,15 @@ public void execute() {
         Collection<SW360Release> sw360ReleasesNotApproved = getNonApprovedReleasesFromSpareReleases(sw360SparseReleases, headers);
 
         List<Artifact> artifacts = sw360ReleasesNotApproved.stream()
-                .map(release -> SW360ReleaseAdapterUtils.convertToArtifact(release, new Artifact("SW360")))
+                .map(release -> {
+                    Artifact artifact = SW360ReleaseAdapterUtils.convertToArtifactWithoutSourceFile(release, new Artifact("SW360"));
+                    Set<SW360SparseAttachment> sparseAttachments = getSparseAttachmentsSource(release);
+                    sparseAttachments.forEach(sparseAttachment -> {
+                        Optional<Path> path = connectionConfiguration.getSW360ReleaseClientAdapter().downloadAttachment(release, sparseAttachment, configuration.getSourcesPath(), headers);
+                        path.map(pth -> artifact.addFact(new ArtifactSourceFile(pth)));
```

Comment:
Better use path.ifPresent(); map() is more to apply a function to the value which does not have a side effect.